### PR TITLE
V6 update to config /5

### LINKED ISSUE #23179: 6.0 : configuration
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0eb4d762-3910-2c57-8741-3306c15537b6
* Version Independent ID: ba2254a0-ea8e-9b28-e94b-edd0fc58c043
* Content: [Configuration in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-5.0)
* Content Source: [aspnetcore/fundamentals/configuration/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/configuration/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes  #23179

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-6.0&branch=pr-en-us-23188)

### CODE REVIEW

File: aspnetcore/fundamentals/configuration/index/samples/6.x/ConfigSample/Program.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using ConfigSample.Options;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRazorPages();
+
+//var Configuration = 
+
+//builder.Services.Configure<PositionOptions>(Configuration.GetSection(
+//                                                PositionOptions.Position));
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/configuration/index/samples/6.x/ConfigSample/Program.cs:11]: @halter73  how do I get `Configuration` here? We need to add that to [Migrate from ASP.NET Core 5.0 to 6.0](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60?view=aspnetcore-5.0&tabs=visual-studio)

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/configuration/index/samples/6.x/ConfigSample/Program.cs:11]: @halter73  need to migrate the following too:
```
public void ConfigureServices(IServiceCollection services)
{
    services.Configure<PositionOptions>(
        Configuration.GetSection(PositionOptions.Position));
    services.Configure<ColorOptions>(
        Configuration.GetSection(ColorOptions.Color));

    services.AddScoped<IMyDependency, MyDependency>();
    services.AddScoped<IMyDependency2, MyDependency2>();

    services.AddRazorPages();
}
```

Comment (halter73) [reply aspnetcore/fundamentals/configuration/index/samples/6.x/ConfigSample/Program.cs:11]: This should work

```suggestion
builder.Services.Configure<PositionOptions>(
	builder.Configuration.GetSection(PositionOptions.Position));
```

`WebApplicationBuilder.Configuration` is a `ConfigurationManager` which is a new type introduced in .NET 6 that implements both `IConfigurationBuilder` and `IConfiguration`, so you can both add config sources and read from it.

File: aspnetcore/fundamentals/configuration/index/samples/6.x/ConfigSample/Program.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+app.Run();
+#elif Third
+#region snippet3
+using Microsoft.Extensions.DependencyInjection.ConfigSample.Options;
+
+var builder = WebApplication.CreateBuilder(args);
+
+
+builder.Services.AddConfig(/* Need Configuration here */ )
+         .AddMyDependencyGroup();
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/configuration/index/samples/6.x/ConfigSample/Program.cs:99]: @halter73 I think this is the last one not working

Comment (halter73) [inline aspnetcore/fundamentals/configuration/index/samples/6.x/ConfigSample/Program.cs:99]: This works, right?

```suggestion
builder.Services.AddConfig(builder.Configuration)
    .AddMyDependencyGroup();
```

**Pull request merged**